Makefile

```diff
@@ -1,4 +1,17 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+ifndef BUILD_DIRECTORY
```

Comment:
This doesn't make sense - `BUILD_DIRECTORY` will **always** be defined.
I guess we could check that `$(dirname $(BUILD_DIRECTORY))` is actually a directory that exists?
FYI: `$(dirname release)` == `.`